src/weapon/w_003.c

```diff
@@ -3,7 +3,59 @@
 #include "weapon_private.h"
 #include "shared.h"
 
-INCLUDE_ASM("weapon/nonmatchings/w_003", EntityWeaponAttack);
+extern WeaponAnimation D_19000_8017AC28[];
+
+void EntityWeaponAttack(Entity* self) {
+    WeaponAnimation* anim;
+    s8 subType;
```

Comment:
How did you pick the name `subType` for this? I'm not sure what the `params` actually controls. For some entities, it is a subtype, but for others it might control other things like whether an entity is in an "angry" state, or HP values, or color, or whatever else. In this case, it looks like ti selects the animation? I might call it animIndex or something. Would be good if we could determine where the `params` are coming from to really be able to tell how the animations are selected.

Reply:
`subType` was just taken from other similar functions. I haven't gone through and triggered any of these to see what they were doing.

Reply:
I've renamed this `animIndex` to make it more generic.